### PR TITLE
Add OpenMC Plotter

### PR DESCRIPTION
**Pull request by kkiesling**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by paulromano** [comment]
I consent to being a maintainer on this recipe.

**Comment by kkiesling** [comment]
confusing behavior to me. It builds for osx, but `openmc-plotter --help` gives this error. Running the `--help` should not prompt any loading of any `.xml` files.
Screen Shot 2022-05-04 at 2 25 22 PM

And for the linux build, I can't seem to get past [this](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=501888&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1790) `libGL` error at runtime.

**Comment by kkiesling** [comment]
@conda-forge/help-python - I am trying to include the `tests` folder that is in the source repository for the purpose of testing, but it not finding the folder. I included `tests` in the `source_files` section, yet this error is appearing. Please advise on the correct way to do this. Thanks!

File: recipes/openmc-plotter/meta.yaml
Comment (carterbox) [inline recipes/openmc-plotter/meta.yaml:17]: What is the reason that this package cannot be noarch: python? It is noarch on PyPI.

File: recipes/openmc-plotter/meta.yaml
Comment (carterbox) [inline recipes/openmc-plotter/meta.yaml:35]: The tests folder isn't included in the PYPI source release, that's why conda-build can't copy it. Switch to the GitHub release url. https://pypi.org/project/openmc-plotter/#files

**Review by carterbox — CHANGES_REQUESTED** [review]
Looks like you figured out that yum_requirements is required for opengl to be available. Good.

File: recipes/openmc-plotter/meta.yaml
Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml:35]: Thank you!

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:10]: This should be the git release, not the actual repo: https://github.com/openmc-dev/plotter/archive/refs/tags/v0.2.0.tar.gz

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml]: If a dependency is missing, it should still be noarch as the dependency can be made available later. Then, the user can install the package without you having to change anything.

**Comment by kkiesling** [comment]
@paulromano - including `pytest.ini` did prompt the tests, but the still failed. I cannot tell what the issue is and adding `-v` to pytest did not provide any additional information for debugging. Thoughts on just using the `python -c "import openmc_plotter"` as a test?

**Comment by paulromano** [comment]
@kkiesling I'm fine with that!

**Comment by kkiesling** [comment]
@pshriwise you just need to consent to being a maintainer here still and then I can submit this for official review.

**Comment by pshriwise** [comment]
Thanks for the reminder @kkiesling.

I consent to being a maintainer on this recipe.

**Comment by kkiesling** [comment]
@conda-forge/help-python, ready for review!

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:17]: Is the only reason you are skipping it a missing dependency? In that case it can be noarch so that users can install this package on windows, once the missing dependency becomes available.

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:36]: This is the same as the `imports` part above so it can be removed.

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:29]: I cannot see this upstream?

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:9]: Since you're not running the tests anymore, can this be PyPI again?

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, then this is ready to go 🚀

File: recipes/openmc-plotter/meta.yaml
Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml:17]: @paulromano can you weigh in on this? If I don't skip Windows it tries to install OpenMC which is not available through conda-forge, which leads to a failure. This OpenMC plotter package is from the same org as OpenMC that causes this issue.

Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml:17]: If it fails because of the missing dependency it doesn't matter. Then it can be noarch.

Ref: Third note here: https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=noarch#noarch-python

Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml]: This is the [issue](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=514492&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=808) that occurs if I add `noarch: python` instead of `skip: true  # [win]`. What's the correct way to handle this error?

Comment (ndevenish) [reply recipes/openmc-plotter/meta.yaml]: The CI isn't loading properly for me so I can't see it, but if it's just the windows build failing because the dependency is not available, _it's fine for the build to fail in this PR for that reason_ - the actual noarch package from the feedstock is done on linux only, so it'll pass the build - it just won't be installable on windows, unless the upstream package becomes available (e.g. IF OpenMC were to add a windows build, you would presumably want this package to just work without changing it).

File: recipes/openmc-plotter/meta.yaml
Comment (ndevenish) [reply recipes/openmc-plotter/meta.yaml]: This comes from somewhere in `openmc`, which [doesn't](https://github.com/conda-forge/openmc-feedstock/blob/926cb818dc80533c95d0c46f0fc4261efe784179/recipe/meta.yaml#L102-L110) run `pip check`

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml]: @ndevenish Is Azure also not loading old logs for you? I’ve been seeing that a couple if times now and wasn’t sure if it’s something on my side.

Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml]: Yes, it was just windows failing in CI for that reason, so I will let that go then.

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml]: You can make a PR there and add it here until it is merged. Just add a link to the PR to the meta.yaml here as a note.

Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml]: Thanks for making that PR @ndevenish - I will merge that and then I think this would be ready to go after that trickles through.

Comment (ndevenish) [reply recipes/openmc-plotter/meta.yaml]: Oh, haha, I didn't clock that it was you for both repos - NP

Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml]: > If there are issues merging, then it's probably okay to either a) include ipython, or b) not run pip check - I don't know what the staged-recipes maintainers preference would be on that.

I always prefer to have a working recipe so if something else is broken, make a PR to fix it there and add it here until the other PR is merged. Better to add something than to ship something that's broken. :)

**Comment by BastianZim** [comment]
Is this ready for review then?

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:21]:
```suggestion
    - python  >=3.6
```
I think GitHub Actions and therefore the linter are down.

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:33]:
```suggestion
    - pip check
    - openmc-plotter --help
```
To check the entry point.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just some small things left.

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml:33]: Ahh ok. Was it one of the following errors? I cannot see the logs anymore for some reason.

https://conda-forge.org/docs/maintainer/maintainer_faq.html#mfaq-qt-load-xcb

Comment (BastianZim) [reply recipes/openmc-plotter/meta.yaml:35]: Interesting 🤔
Then please revert the CLI test and I’ll merge it.

Comment (kkiesling) [reply recipes/openmc-plotter/meta.yaml:35]: I will revert it- the issue is actually just with argparse, not qt. It seems to go deeper into the code with `--help` than normal on CI because it is looking for input files to be run even with just `--help` (input files that are required from the user when the tool is used).

File: recipes/openmc-plotter/meta.yaml
Comment (BastianZim) [inline recipes/openmc-plotter/meta.yaml:17]: Please also remove this again

**Pull request merged**
